crates/telio-wg/src/wg.rs

```diff
@@ -747,9 +873,72 @@ impl State {
         }
 
         self.interface = to.clone();
-
         success
     }
+
+    fn compute_new_node_info(
+        &self,
+        old_info: Option<&NoLinkDetectionInfo>,
+        wg_peer: &Peer,
+    ) -> Option<NoLinkDetectionInfo> {
+        let configurable_rtt = match self.no_link_detection {
+            NoLinkDetection::Disabled => DEFAULT_CONFIGURABLE_RTT,
+            NoLinkDetection::Enabled { rtt } => rtt,
+        };
+
+        let (new_tx, new_rx) = match (wg_peer.tx_bytes, wg_peer.rx_bytes) {
+            (Some(tx_bytes), Some(rx_bytes)) => (tx_bytes, rx_bytes),
+            _ => {
+                telio_log_warn!(
+                    "Number of bytes missing from wg peer {}",
+                    wg_peer.public_key
+                );
+                return None;
+            }
+        };
+
+        match old_info {
+            Some(old_info) => {
+                let now = Instant::now();
+
+                let (rx_bytes, rx_ts) = if new_rx > old_info.last_rx_bytes {
+                    (new_rx, now)
+                } else {
+                    (old_info.last_rx_bytes, old_info.last_rx_timestamp)
+                };
+
+                let (tx_bytes, tx_ts) = if new_tx > old_info.last_tx_bytes {
+                    // We should not bother about passive keepalives, they will not modify
+                    // the tx_bytes count received from wg
```

Comment:
Hmmm.. Is this actually true? Sending the keepalive(passive or persistent) does not increase `tx_bytes`? The payload is null however the headers still take up space. Without detecting the keepalive which is outgoing we will falsely assume that we need to wait for a response where we don't actually